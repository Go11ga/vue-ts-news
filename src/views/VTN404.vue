<template>
    <div class="vtn-404">
        <div class="vtn-404__code">
            404
        </div>

        <div class="vtn-404__text">
            Страница не найдена
        </div>

        <router-link class="vtn-404__link" to="/">
            На главную
        </router-link>
    </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { mapMutations } from 'vuex'

export default defineComponent({
    name: 'VTN404',

    methods: {
        ...mapMutations([
            'SET_ACTIVE_CHANNEL'
        ])
    },

    mounted() {
        this.SET_ACTIVE_CHANNEL('')
    }
})
</script>
