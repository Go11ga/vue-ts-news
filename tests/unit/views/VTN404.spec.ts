import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'
import VTN404 from '@/views/VTN404.vue'

describe('VTN404', () => {
    let wrapper: any
    const mutations = {
        SET_ACTIVE_CHANNEL: jest.fn()
    }

    const store = createStore({
        mutations
    })

    wrapper = shallowMount(VTN404, {
        global: {
            stubs: ['router-link'],
            plugins: [store]
        }
    })

    it('dummy name test', () => {
        expect(wrapper.vm.$options.name).toMatch('VTN404')
    })

    it('snapshot', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

    it('хук mutation', () => {
        expect(typeof VTN404.mounted).toBe('function')

        wrapper.vm.SET_ACTIVE_CHANNEL = jest.fn()

        wrapper.vm.$options.mounted.call(wrapper.vm)
        
        expect(wrapper.vm.SET_ACTIVE_CHANNEL).toBeCalledWith('')
    })
})
