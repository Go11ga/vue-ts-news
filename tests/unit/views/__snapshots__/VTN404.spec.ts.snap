// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTN404 snapshot 1`] = `
<div
  class="vtn-404"
>
  <div
    class="vtn-404__code"
  >
     404 
  </div>
  <div
    class="vtn-404__text"
  >
     Страница не найдена 
  </div>
  <router-link-stub
    class="vtn-404__link"
    to="/"
  />
</div>
`;
